internal/transport/transport_test.go

```diff
@@ -536,6 +536,52 @@ func (s) TestInflightStreamClosing(t *testing.T) {
 	}
 }
 
+// tests that when streamID > MaxStreamId, the current client transport drains.
+func (s) TestClientTransportDrainsAfterStreamIDExhausted(t *testing.T) {
+	server, ct, cancel := setUp(t, 0, math.MaxUint32, normal)
+	defer cancel()
+	defer server.stop()
+	callHdr := &CallHdr{
+		Host:   "localhost",
+		Method: "foo.Small",
+	}
+	// override MaxStreamID.
+	originalMaxStreamID := MaxStreamID
+	MaxStreamID = 3
+	defer func() {
+		MaxStreamID = originalMaxStreamID
+	}()
+
+	ctx, ctxCancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer ctxCancel()
+
+	s, err := ct.NewStream(ctx, callHdr)
+	if err != nil {
+		t.Fatalf("ct.NewStream() = %v", err)
+	}
+	if s.id != 1 {
+		t.Fatalf("stream id: %d, want: 1", s.id)
+	}
+
+	if err := ct.compareStateForTesting(reachable); err != nil {
+		t.Fatal(err)
+	}
+
+	s, err = ct.NewStream(ctx, callHdr)
```

Comment:
Maybe add a comment here mentioning that a single stream takes up two IDs, which is why the next stream ID shows up as `3` here.